File: 品味Java的21種設計模式/dp_cpp/abstractfactory/example1/AMDCPU.cpp
#include "AMDCPU.h"
#include <iostream>

namespace cn {
namespace javass {
namespace dp {
namespace abstractfactory {
namespace example1 {

AMDCPU::AMDCPU(int pins) {
    this->pins = pins;
}

void AMDCPU::calculate() {
    std::cout << "now in AMD CPU,pins=" << pins;
}
}
}
}
}
}

File: 品味Java的21種設計模式/dp_java/abstractfactory/example1/AMDCPU.java
package cn.javass.dp.abstractfactory.example1;
/**
 * AMD的CPU实现
 */
public class AMDCPU implements CPUApi {
    /**
     * CPU的针脚数目
     */
    private int pins = 0;
    /**
     * 构造方法，传入CPU的针脚数目
     * @param pins CPU的针脚数目
     */
    public AMDCPU(int pins) {
        this.pins = pins;
    }
    public void calculate() {
        System.out.println("now in AMD CPU,pins=" + pins);
    }
}
